crates/storage/src/store/multistore.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::Arc;
+
+use super::substore::SubstoreConfig;
+
+/// A collection of substore, each with a unique prefix.
+#[derive(Debug, Clone)]
+pub struct Multistore {
+    transparent_store: Arc<SubstoreConfig>,
+    substores: Vec<Arc<SubstoreConfig>>,
+}
+
+impl Multistore {
+    pub fn new(mut substores: Vec<Arc<SubstoreConfig>>) -> Self {
+        assert!(
+            !substores.is_empty(),
+            "multistore must have at least one substore"
+        );
+
+        Self {
+            transparent_store: substores.swap_remove(0),
+            substores,
+        }
+    }
+
+    pub fn _get(&self, prefix: &str) -> Option<Arc<SubstoreConfig>> {
+        if prefix.is_empty() {
+            Some(self.transparent_store.clone())
+        } else {
+            self.substores.iter().find(|s| s.prefix == prefix).cloned()
+        }
+    }
```

Comment:
What do we do with this? When do we want to return `None` here, isn't the point of the root store that it handles the case where no prefix matches?

Reply:
(I guess this also applies to the `find_substore` method below, though in that case the question is whether we want the caller to have to deal with filling in the root store, or whether the routing method should do that).